ktor-client/ktor-client-cio/src/io/ktor/client/engine/cio/CIOHttpResponse.kt

```diff
@@ -28,16 +30,14 @@ internal class CIOHttpResponse(
 
     override val responseTime: GMTDate = GMTDate()
 
-    override val executionContext: CompletableDeferred<Unit> = CompletableDeferred()
-
     override fun close() {
+        super.close()
         if (pipelined) {
             runBlocking {
                 val length = headers[HttpHeaders.ContentLength]!!.toLong()
                 content.discard(length - content.totalBytesRead)
             }
         }
         response.release()
-        executionContext.complete(Unit)
```

Comment:
Now we have changed completion order. Doesn't it affect anything?

Reply:
Nope. We're still waiting for the child tasks